### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by aueti**
Added documentation for
- Special terms
- Numbers
- User Defined Variables
- Operations
- Conditionals
- For Loops
- Function Calls

### CODE REVIEW

File: Documentation.md [hunk 1]
```diff
@@ -4,20 +4,91 @@
...
+
+| Section Type   | Definition      | Description                                                  | Example       |
+| -------------- | --------------- | ------------------------------------------------------------ | ------------- |
+| Title          | ^(.*?[A-Z]){2,} | The title is made of at least 2 word that are all capitalized | FUNCTION CALL |
+| Internal Logic |                 | If the function is called, this line(s) will be executed.    |               |
+
+When calling a function the structure follows "parameter1, paramter2" (Function Call). The (Function Call) section references the function you made previously. The (Function Call) doesn't need to be capitalized here. The "parameter1, parameter2" indicates the list of parameters, for this example in the Function Call function, parameter1 maps to function and parameter 2 maps to call.
```
Comment (Nick-Kildall) [inline Documentation.md]: Found typo, "paramter2" should be changed to "parameter2"

**Comment by Nick-Kildall** [comment]
Overall, this documentation looks great — thank you for creating this!

File: Documentation.md (hunk 1 above)
Comment (aueti) [reply Documentation.md]: Thanks!

**Comment by worseTyler** [comment]
Much appreciated! People might actually be able to understand how to use our amazing language

**Pull request merged**